doc/layer1_refutation_game.md

```diff
@@ -0,0 +1,216 @@
+- however, even if a block is not refuted, because a game was fixed, or because the challenger made a mistake, *it can still be challenged again*
+    - as long as someone honest is observing the chain, he can challenge, and he will win if he makes no mistake.
+
+## Architecture
+
+The refutation game can be seperated in several parts:
+- initialisation
+    - blocks challenged
+    - bond setting
+- bissection
+    - back and forth between players to find the first opcode disagreement
+- verification
+    - checking of one step proof
+- resolution
+    - deciding on winner
+    - slashing
+    - rejection of block
+
+## Initialisation
+
+- is number of operations fixed in a block ?
+    - if not, it should at least be deterministic, otherwise two truthfull blocks can be in opposition
+        - special case of game, where at the beginning the stacked on longer segment aggrees that opposing block is a prefix ?
+    - take one with the biggest number of op ?
+    - do block mention operations included ?
+        - maybe not, but then finality of an operation in inbox is not known ? is it a pb ? maybe for withdraw
+    - is it a question for the running of the game or the initiation of a game ?
+- how to ensure a block "finishes" all operations ? 
+    - exemple of cheat : normal execution takes 100 steps, but only run 90 steps. The hash are truthfull, but the final state is incorrect in the sense that some operations are not taken into account.
+    - a way to fixe that is for the state to include the stack of all operations to precess, which allows to prove that final hash0 contains empty stack, therefore that all operations were finished. 
+        - a proof of the last state would include the fact that the stack must be empty ?
+        - still need to flag a hash as "should be last state" to know a check is needed
+    - or a block doesn't refer to the list of operations at all
+        - what happens when two honest blocks are proposed, but of different size ?
+        - how does one knows when an operation is processed ?
+    - is it in the definition of "valid block" ? 
+        - a block contains the proof that execution stack is empty
+    - is it a question for the running of the game or the initiation of a game or the verification of proof ?
+
+## Bissection
+
+```
+segment = hash * hash * size
+split = segment * segment
+```
+
+Start: One player proposes/defends a segment of `length > 1`
+End: One player has to provide a proof for segment of `length = 1`
+
+### Simple version of Bissection
+
+`A` propose a segment, `B` challenges it
+`A` either 
+- provides a proof 
+    - proof is checked
+- or proposes 2 sub-segments
+    - must be correct split: same ends, same overall size, all sub-segment are shorter
+    -  `B` chooses which one to challenge
+
+
+**Game 1: simple game (A defends)**
+```graphviz
```

Comment:
Am I missing a way to see this diagrams? My github only shows text

Reply:
I wrote them on hackmd, I'm not sure github allows graphviz :l

Reply:
I'm not sure what to do actually regarding this pb.

Reply:
maybe `mermaid` instead? It works on both hackmd and github.

https://github.com/marigold-dev/chusai/blob/main/doc/layer1_contracts_architecture.md